src/oscar/apps/dashboard/ranges/forms.py

```diff
@@ -33,6 +33,7 @@ class RangeProductForm(forms.Form):
 
     def __init__(self, range, *args, **kwargs):
         self.range = range
+        self.excluded = kwargs.pop('excluded', False)
```

Comment:
If you add a new hidden input field, which describes what kind of form it is, you can just remove this.
Inside the clean method you could just access the hidden form field and do logic based on that.

```python
upload_type = forms.CharField(widget=forms.HiddenInput())
```